rudder-web/src/main/scala/com/normation/rudder/web/services/Section2FieldService.scala

```diff
@@ -223,41 +220,40 @@ class Section2FieldService(val fieldFactory: DirectiveFieldFactory, val translat
     SectionFieldImp(sectionSpec.name, children, priorityToVisibility(sectionSpec.displayPriority), varMappings)
   }
 
-
-  // transforms
-  // Map(A -> Seq("A1", "A2"), B -> Seq("B1", "b2"))
-  // to
-  // Seq( Map((A -> "A1"), (B -> "B1")),
-  //      Map((A -> "A2"), (B -> "B2")) )
-  //If there is no value, a None is returned
+  // Transforms map of all values by Variable Name
+  // Into the Sequence of all values by section
+  // A section will be defined as a Map of value by variable name
+  //   * ie: Map(A -> Seq("A1", "A2"), B -> Seq("B1", "B2"))  => Seq( Map((A -> "A1"), (B -> "B1"))
+  //                                                                , Map((A -> "A2"), (B -> "B2")) )
+  // If there is no value, a Section with default value is returned is returned
   private def createMapForEachSubSection(section: SectionSpec, valuesByName:Map[String,Seq[String]]): Seq[Map[String, Option[String]]] = {
-    // values represent all the values we have for the same name of variable
-    case class NameValuesVar(name: String, values: Seq[String]) extends HashcodeCaching
 
-    // seq of variable values with same name correctly ordered
-    val seqOfNameValues : Seq[NameValuesVar] = {
-      for {
-        varSpec <- section.getAllVariables
-      } yield {
-        NameValuesVar(varSpec.name, valuesByName.getOrElse(varSpec.name, Seq[String]()))
-      }
-    }
+    // Variable with max value determine the actual number of section
+    // If a varaible has 6 values, We need to have 6 sections at the end
+    val numberMaxofSections = valuesByName.map(_._2.size).max
 
-    if (seqOfNameValues.isEmpty) {
-      Seq(Map[String, Option[String]]())
-    } else {
-      for {
-        // If head has an empty sequence as value, it does not iterate for other variables
-        // To fix, we use the max size of of all variables (so those value can be used, missing will be set to None.
-        i <- 0 until seqOfNameValues.map(_.values.size).max
-      } yield {
-        for {
-          nameValues <- seqOfNameValues
+    for {
+      // For each section
+      sectionIndex <- 0 until numberMaxofSections
+
+    } yield {
+      ( for {
+        varSpec <- section.getAllVariables
         } yield {
-          val valueOpt = try Some(nameValues.values(i)) catch { case e: Exception => None }
-          (nameValues.name, valueOpt)
+          val value = valuesByName.get(varSpec.name) match {
+            // No value defined for the variable, get default value
+            case None => varSpec.constraint.default
+            // Some value for this variable, look if defined for this section
+            case Some(values) => values.lift(sectionIndex) match {
+              // No value defined, get default value
+              case None => varSpec.constraint.default
+              // Get value
+              case value => value
+            }
+          }
+          (varSpec.name,value)
         }
-      }.toMap
+      ).toMap
     }
```

Comment:
What about the case where you have a section, which contains a variable A and a multi-valued subsection: in that case, you don't have numberMaxofSections for the A if there is several subsection, and you even can have 0 values for subsection (all optionnal) with numberMaxofSections == 1 (because of A), and you don't want to put defaults values in that case (because the user want 0 subsection, not one with defaults values)?